qdrant_client/local/payload_filters.py

```diff
@@ -86,18 +87,40 @@ def check_geo_polygon(condition: models.GeoPolygon, values: Any) -> bool:
     return False
 
 
+def check_range_interface(condition: models.RangeInterface, value: Any) -> bool:
+    if isinstance(condition, models.Range):
+        return check_range(condition, value)
+    if isinstance(condition, models.DatetimeRange):
+        return check_datetime_range(condition, value)
+    return False
+
+
 def check_range(condition: models.Range, value: Any) -> bool:
     if not isinstance(value, (int, float)):
         return False
-    if condition.lt is not None and value >= condition.lt:
-        return False
-    if condition.lte is not None and value > condition.lte:
-        return False
-    if condition.gt is not None and value <= condition.gt:
+    return (
+        (condition.lt is None or value < condition.lt)
+        and (condition.lte is None or value <= condition.lte)
+        and (condition.gt is None or value > condition.gt)
+        and (condition.gte is None or value >= condition.gte)
+    )
```

Comment:
it might be harder to read, but ok

Reply:
I wanted to keep the same sign for the operator (>= for gte, etc). 😄